src/main/kotlin/org/onflow/flow/sdk/crypto/Crypto.kt

```diff
@@ -170,11 +193,23 @@ internal class SignerImpl(
     override val hasher: Hasher = HasherImpl(hashAlgo)
 ) : Signer {
     override fun sign(bytes: ByteArray): ByteArray {
-        val ecdsaSign = Signature.getInstance(hashAlgo.id)
-        ecdsaSign.initSign(privateKey.key)
-        ecdsaSign.update(bytes)
+        val signature: ByteArray
+
+        if (hashAlgo == HashAlgorithm.KECCAK256 || hashAlgo == HashAlgorithm.KMAC128) {
+            // Handle Keccak-256 and KMAC128 separately
+            val hash = hasher.hash(bytes)
+            val ecdsaSign = Signature.getInstance("NONEwithECDSA")
+            ecdsaSign.initSign(privateKey.key)
+            ecdsaSign.update(hash)
+            signature = ecdsaSign.sign()
+        } else {
+            // Handle other algorithms with valid IDs
+            val ecdsaSign = Signature.getInstance(hashAlgo.id)
```

Comment:
- here I see how `hashAlgo.id` is used. I still think the string shouldn't be stored in `hashAlgo` as it is related to signing only and not hashing. We could have a switch-case statement where the input string to `initSign` is set. Another option is to hash first and then sign the hash with `Signature.getInstance("NONEwithECDSA")`, without separating the two cases.
- for consistency with the [go-sdk](https://github.com/onflow/flow-go-sdk/blob/master/crypto/crypto.go#L68-L72), we can disable ECDSA with sha2-384 and sha3-384. It makes sense to enable those if we support 384 bits curve but that's not the case today.

Reply:
@tarakby thank you for the feedback and comments, will review these today and adjust! Thanks!

Reply:
Disabled ECDSA with SHA2-384 and SHA3-384